### PR TITLE
introduce Infinity as atom to handle division by 0

### PR DESCRIPTION
**Pull request by pubrrr**

### CODE REVIEW

File: src/domain/expression_tree.rs [hunk 1]
```diff
@@ -94,70 +97,220 @@ impl Evaluation<f64> for Expression {
...
     }
 }
 
+impl Add for Atom {
+    type Output = Atom;
+
+    fn add(self, other: Self) -> Self::Output {
+        match (self, other) {
+            (NegativeInfinity, NegativeInfinity) => NegativeInfinity,
+            (Infinity, _) | (_, Infinity) | (NegativeInfinity, _) | (_, NegativeInfinity) => Infinity,
```
Comment (ekinimo) [inline src/domain/expression_tree.rs:133]: -infinity + x = infinity !!!??

File: src/domain/expression_tree.rs [hunk 2]
```diff
@@ -94,70 +97,220 @@ impl Evaluation<f64> for Expression {
...
+        }
+    }
+}
+
+impl Sub for Atom {
+    type Output = Atom;
+
+    fn sub(self, other: Self) -> Self::Output {
+        match (self, other) {
+            (Infinity, _) | (_, Infinity) | (NegativeInfinity, _) | (_, NegativeInfinity) => Infinity,
```
Comment (ekinimo) [inline src/domain/expression_tree.rs:147]: same problem

File: src/domain/expression_tree.rs [hunk 3]
```diff
@@ -94,70 +97,220 @@ impl Evaluation<f64> for Expression {
...
+    }
+}
+
+impl Mul for Atom {
+    type Output = Atom;
+
+    fn mul(self, other: Self) -> Self::Output {
+        match (self, other) {
+            (Infinity, NegativeInfinity) | (NegativeInfinity, Infinity) => NegativeInfinity,
+            (Infinity, _) | (_, Infinity) | (NegativeInfinity, _) | (_, NegativeInfinity) => Infinity,
```
Comment (ekinimo) [inline src/domain/expression_tree.rs:162]: also here

File: src/domain/expression_tree.rs [hunk 4]
```diff
@@ -94,70 +97,220 @@ impl Evaluation<f64> for Expression {
...
+        }
+    }
+}
+
+impl Div for Atom {
+    type Output = Atom;
+
+    fn div(self, other: Self) -> Self::Output {
+        match (self, other) {
+            (Infinity, NegativeInfinity) | (NegativeInfinity, Infinity) => NegativeInfinity,
```
Comment (ekinimo) [inline src/domain/expression_tree.rs:176]: mazbe introduce NaN. infinity/infinity is not necerraly infinity

**Review by ekinimo — CHANGES_REQUESTED** [review]
also maybe fix infinity in parser ? if we cant parse the result because its too big simply return infinity ?

**Comment by pubrrr** [comment]
This solution does not seem to bring much benefit...

**Pull request closed without merging**